From 6aa1b4fbf5936a1ff5bdbb79397c94910a6ed8f5 Mon Sep 17 00:00:00 2001
From: Anna Bauza <annabauza@google.com>
Date: Fri, 2 Aug 2024 09:06:32 +0000
Subject: [PATCH] fix: Security Report - Reveal images across users via
 EditUserPhotoController

This functionality has implemented tests on t+ branches.

Bug: 296915959
Test: N/A
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:994041531919f780dcd4d3cbf9801eb4f76dcea7)
Merged-In: If79af734432b14be74815a47e1026dc8369a304f
Change-Id: If79af734432b14be74815a47e1026dc8369a304f
---
 .../android/settingslib/users/EditUserPhotoController.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/packages/SettingsLib/src/com/android/settingslib/users/EditUserPhotoController.java b/packages/SettingsLib/src/com/android/settingslib/users/EditUserPhotoController.java
index 0c6cd048619c..1c3dea71ad45 100644
--- a/packages/SettingsLib/src/com/android/settingslib/users/EditUserPhotoController.java
+++ b/packages/SettingsLib/src/com/android/settingslib/users/EditUserPhotoController.java
@@ -18,6 +18,7 @@ package com.android.settingslib.users;
 
 import android.app.Activity;
 import android.content.ClipData;
+import android.content.ContentProvider;
 import android.content.ContentResolver;
 import android.content.Context;
 import android.content.Intent;
@@ -140,6 +141,12 @@ public class EditUserPhotoController {
             return false;
         }
 
+        final int currentUserId = UserHandle.myUserId();
+        if (currentUserId != ContentProvider.getUserIdFromUri(pictureUri, currentUserId)) {
+            Log.e(TAG, "Invalid pictureUri: " + pictureUri);
+            return false;
+        }
+
         switch (requestCode) {
             case REQUEST_CODE_CROP_PHOTO:
                 onPhotoCropped(pictureUri);
-- 
2.46.1.824.gd892dcdcdd-goog

